datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +50,26 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
+
+    Map<String, PartitionThroughputInfo> throughputMap = throughputInfo.getPartitionInfoMap();
+
+    Set<String> allPartitions = new HashSet<>(assignedPartitions);
+    allPartitions.addAll(unassignedPartitions);
+
+    // total throughput in KB/sec
+    int totalThroughput = allPartitions.stream().map(p ->
+        throughputMap.getOrDefault(p, new PartitionThroughputInfo(BYTES_IN_KB_RATE_DEFAULT, MESSAGES_IN_RATE_DEFAULT, p))).
+        mapToInt(PartitionThroughputInfo::getBytesInKBRate).sum();
+    LOG.info("Total throughput in all {} partitions: {}KB/sec", allPartitions.size(), totalThroughput);
+
+    double taskCapacityUtilizationCoefficient = _taskCapacityUtilizationPct / 100.0;
+    int taskCountEstimate = (int) Math.ceil((double) totalThroughput /
+        (_taskCapacityMBps * 1024 * taskCapacityUtilizationCoefficient));
```

Comment:
nit: if you save `_taskCapacityMBps * 1024 * taskCapacityUtilizationCoefficient` calculation, you can get rid of `_taskCapacityMBps`, `taskCapacityUtilizationCoefficient` and extra calculation.

Reply:
Yeah but then I'll need to name that variable 😄